lib/tandem/tandemSimulator.js

```diff
@@ -79,6 +79,22 @@ exports.make = function(config){
 
   return {
     alarm: function(event) {
+      if (currStatus !== null && currStatus.status === 'suspended') {
+        var type = event.alarmType;
+        if(type === 'occlusion' || type === 'auto_off' || type === 'no_insulin' || type === 'no_power' ||
+            (type === 'other' && event.payload.alarm_name === 'Cartridge Removed') ) {
+
+          var status = currStatus;
+          status.reason = {suspended: 'automatic'};
+          if(type === 'other') {
+            status.payload = {cause: type, description: event.payload.alarm_name};
```

Comment:
Sadly, probably need to do the check for `isAssigned('payload')` and all that here too. The logIndices for the `suspended` are being steamrolled.